### DISCUSSION TITLE
Ability to only `extend-horizontally` for `window-padding-color`

### DISCUSSION THREAD

**Discussion by zuqini** [ghostty-org/ghostty#3357] [Feature Requests, Ideas]
Sometimes setting `window-padding-color = extend` will extend the padding colors vertically which (in my opinion) looks off, but extending horizontally looks fine. See below examples.
_Example of padding extended both vertically and horizontally:_
<img width="997" alt="image" src="https://github.com/user-attachments/assets/41d21a1f-cdfc-4025-a334-cec54daa35d6" />
_Example of padding only extended horizontally:_
<img width="960" alt="image" src="https://github.com/user-attachments/assets/0191ac6a-63cb-4c0e-99af-0b7eacb64976" />

I think adding an option to only extend horizontally might be worthwhile!

**Comment by 00-kat** [comment]
I'm certain someone will want to only extend it vertically but not horizontally, so I think that `window-padding-color` should accept **both** `extend-horizontal`/`extend-horizontal-always` and `extend-vertical`/`extend-vertical-always`.

**Comment by pluiedev** [comment]
Perhaps we should split `window-padding-color` into horizontal and vertical variants? (Or, go down the `theme` route and support `horizontal:` and `vertical:` prefixes) Need more design passes on that idea.

**Reply by 00-kat** [reply]
I'm for splitting the option.

> Or, go down the `theme` route and support `horizontal:` and `vertical:` prefixes

Most other similar options split it or use prefixes on the values (as I had suggested); in fact `theme` is the only option which uses that syntax right now.  Whether newer configuration options should be changed over to the `theme` syntax sounds like it would be outside the scope of this discussion and perhaps warrants a discussion of its own (discussion on this option could maybe be paused until a hypothetical new discussion is closed?).

**Comment by krh** [comment]
I'd also suggest `clear-color` as an option, to pad to the color the terminal was last cleared to. The heuristic of padding the bg color of the nearest character cell, verticaly, horizonally or both often gives odd results. For applications that themes bg and fg colors, padding to the most recent clear color at least consistently picks a color that matches the color theme.

**Reply by 00-kat** [reply]
Can you please [open another Discussion] for that instead of stealing this one?

[open another Discussion]: https://github.com/ghostty-org/ghostty/discussions/new?category=feature-requests-ideas
